src/main/java/bio/terra/service/duos/DuosService.java

```diff
@@ -148,14 +154,112 @@ private void updateLastSynced(DuosFirecloudGroupModel firecloudGroup, Instant la
     try {
       duosDao.updateFirecloudGroupLastSyncedDate(id, lastSyncedDate);
     } catch (CannotSerializeTransactionException ex) {
-      String msg =
-          "Members of Firecloud group %s were updated, "
-              + "but could not update DUOS Firecloud Group record %s with last_synced_date %s"
-                  .formatted(firecloudGroup.getFirecloudGroupName(), id, lastSyncedDate);
+      String message =
+          firecloudGroup.getFirecloudGroupEmail()
+              + " members were updated, "
+              + "but an error occurred when updating its database record with last_synced_date "
+              + lastSyncedDate;
       String errorDetail =
           "This indicates that another process was updating the database row at the same time";
-      throw new DuosFirecloudGroupUpdateConflictException(msg, ex, List.of(errorDetail));
+      throw new DuosFirecloudGroupUpdateConflictException(message, ex, List.of(errorDetail));
+    }
+  }
+
+  /**
+   * @param id of the DUOS Firecloud Group record if it was successfully synced, null otherwise
+   * @param error encountered if the DUOS Firecloud Group failed to sync, null otherwise
+   */
+  record SyncResult(UUID id, ErrorModel error) {}
+
+  /**
+   * For all DUOS dataset IDs registered in TDR, force a sync of their Firecloud group members. A
+   * failure to sync any Firecloud group(s) should not block attempts to sync the rest.
+   *
+   * @return DUOS Firecloud groups whose members were successfully synced, and any errors occurred
+   *     which may have interfered with syncing
+   */
+  public DuosFirecloudGroupsSyncResponse syncDuosDatasetsAuthorizedUsers() {
+    // 1. Parallelize our calls to external systems (DUOS, Sam) which perform the sync
+    Instant lastSyncedDate = Instant.now();
+    List<Future<SyncResult>> futures =
+        duosDao.retrieveFirecloudGroups().stream()
+            .map(group -> executor.submit(() -> syncFirecloudGroupContents(group)))
+            .toList();
+    List<SyncResult> results = FutureUtils.waitFor(futures);
+
+    // 2. Record successful syncs to the DB in a single call to avoid deadlocks
+    List<UUID> syncedIds = results.stream().map(SyncResult::id).filter(Objects::nonNull).toList();
+    Optional<ErrorModel> maybeDbError = batchUpdateLastSynced(syncedIds, lastSyncedDate);
+
+    // 3. Collect any errors emitted along the way.
+    List<ErrorModel> errors =
+        results.stream()
+            .map(SyncResult::error)
+            .filter(Objects::nonNull)
+            .collect(Collectors.toCollection(ArrayList::new));
```

Comment:
Can you just do toList() here? 
```suggestion
        results.stream()
            .map(SyncResult::error)
            .filter(Objects::nonNull)
            .toList();
```

Reply:
Good question!  That would create an immutable list, but we may supplement the error list with any error encountered in our DB update.  So it needs to be mutable.  I'm open to writing this differently though.

Reply:
You can't use `toList()` but you could use `collect(Collectors.toList())`.

Reply:
I'm curious...why can't you use toList?

Reply:
@nmalfroy here is the reasoning re: your comment "I'm curious...why can't you use toList?" (I couldn't reply to it for some reason!)

Reply:
@pshapiro4broad I am not sure we can use `collect(Collectors.toList())` if we need to guarantee mutability.  Here's what I found in the docs:

https://docs.oracle.com/en/java/javase/14/docs/api/java.base/java/util/stream/Collectors.html#toList()

>There are no guarantees on the type, mutability, serializability, or thread-safety of the List returned; if more control over the returned List is required, use [toCollection(Supplier)](https://docs.oracle.com/en/java/javase/14/docs/api/java.base/java/util/stream/Collectors.html#toCollection(java.util.function.Supplier)).

Reply:
oooh ok cool